mlflow/tracking/artifact_utils.py

```diff
@@ -72,3 +77,34 @@ def _download_artifact_from_uri(artifact_uri, output_path=None):
 
     return get_artifact_repository(artifact_uri=root_uri).download_artifacts(
         artifact_path=artifact_path, dst_path=output_path)
+
+
+def _upload_artifacts_to_databricks(source, run_id, source_host_uri=None,
+                                    target_databricks_profile_uri=None):
+    """
+    Copy the artifacts from ``source`` to the destination Databricks workspace (DBFS) given by
+    ``databricks_profile_uri`` or the current tracking URI.
+    :param source: Source location for the artifacts to copy.
+    :param run_id: Run ID to associate the artifacts with.
+    :param source_host_uri: Specifies the source artifact's host URI (e.g. Databricks tracking URI)
+        if applicable. If not given, defaults to the current tracking URI.
+    :param target_databricks_profile_uri: Specifies the destination Databricks host. If not given,
+        defaults to the current tracking URI.
+    :return: The DBFS location in the target Databricks workspace the model files have been
+        uploaded to.
+    """
+    import uuid
+    local_dir = tempfile.mkdtemp()
+    try:
+        # TODO: might want to throw if `source` is a models URI??
+        source_with_profile = add_databricks_profile_info_to_artifact_uri(source, source_host_uri)
+        _download_artifact_from_uri(source_with_profile, local_dir)
+        dest_root = 'dbfs:/databricks/mlflow/tmp-external-source/'  # TODO: "/" or "-"?
+        dest_repo = DbfsRestArtifactRepository(dest_root, target_databricks_profile_uri)
+        dest_dir = run_id if run_id else uuid.uuid1()
+        dest_repo.log_artifacts(local_dir, artifact_path=dest_dir)
+        return dest_root + dest_dir  # new source
+    finally:
+        shutil.rmtree(local_dir)
+    # NOTE: we can't easily delete the target temp location due to the async nature
```

Comment:
we may want to print out the location of the temp file to delete and the command to do so (also add prints showing progress since this fn could take a while)